packages/aws-cdk-lib/aws-cloudwatch/lib/alarm.ts

```diff
@@ -169,11 +218,48 @@ export class Alarm extends AlarmBase {
     addConstructMetadata(this, props);
 
     const comparisonOperator = props.comparisonOperator || ComparisonOperator.GREATER_THAN_OR_EQUAL_TO_THRESHOLD;
+    const isAnomalyDetection = isAnomalyDetectionOperator(comparisonOperator);
+
+    let threshold: number | undefined; // For a literal threshold value
+    let thresholdMetricId: string | undefined; // For anomaly detection
+    if (isAnomalyDetection) {
+      if (!isAnomalyDetectionMetric(props.metric)) {
+        throw new ValidationError(
+          `Anomaly detection operator ${comparisonOperator} requires an ANOMALY_DETECTION_BAND() metric. Use the construct AnomalyDetectionAlarm or wrap your metric in an ANOMALY_DETECTION_BAND expression.`,
+          this,
+        );
+      }
+
+      // For now the property `threshold` is required, so checking it against 'undefined' is a bit of a stretch, but reasonable.
```

Comment:
Since `threshold` is required, should we have a validation error for missing threshold?